cider/type/plan/Analyzer2.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "exec/nextgen/jitlib/JITLib.h"
+#include "exec/nextgen/jitlib/base/JITFunction.h"
+#include "exec/template/Execute.h"
+#include "type/plan/Analyzer.h"
+
+namespace Analyzer {
+using namespace cider::jitlib;
+
+JITTypeTag Expr::getJITTag(const SQLTypes& st) {
```

Comment:
Maybe we could separate this func to ```nextgen/utils```? So other components could use it.

Reply:
only Expr needs this so far. 
We can make this change any time when we found others need this.